src/utils.ts

```diff
@@ -755,6 +763,33 @@ export function getWebViewFavicon(extensionPath: string): vscode.Uri {
   );
 }
 
+export async function createNewComponent(
+  name: string,
+  currentDirectory: string
+) {
+  const componentDirPath = path.join(currentDirectory, "components", name);
+  await mkdirp(componentDirPath);
+  const newComponentTemplatePath = path.join(
+    extensionContext.extensionPath,
+    "templates",
+    "new_component"
+  );
+  await copy(newComponentTemplatePath, componentDirPath);
+  const header = path.join(componentDirPath, "include", "new_component.h");
+  const newHeader = path.join(componentDirPath, "include", `${name}.h`);
+  await move(header, newHeader);
+  const sourceFile = path.join(componentDirPath, "new_component.c");
+  const newSourceFile = path.join(componentDirPath, `${name}.c`);
+  await move(sourceFile, newSourceFile);
```

Comment:
maybe create a closure to do this repeat functionality ?

```suggestion
async function rename(oldName: string, newName: string) {
  const oldPath = path.join(componentDirPath, "include", oldName);
  const newPath = path.join(componentDirPath, "include", newName);
  await move(header, newHeader);
}
await rename("new_component.h", `${name}.h`)
await rename("new_component.c", `${name}.c`)
```